verifiers/internal/gha/slsaprovenance/v1.0/byob_test.go

```diff
@@ -0,0 +1,259 @@
+					},
+				},
+			},
+			branch: "",
+		},
+	}
+
+	for i := range testCases {
+		tt := testCases[i]
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			branch, err := tt.prov.GetBranch()
+			if diff := cmp.Diff(tt.err, err, cmpopts.EquateErrors()); diff != "" {
+				t.Fatalf("unexpected error: %v", err)
+			}
+			if got, want := branch, tt.branch; got != want {
+				t.Fatalf("unexpected branch, got: %q, want: %q", got, want)
+			}
+		})
+	}
+}
+
+func Test_BYOBProvenance_GetTag(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name string
+		prov BYOBProvenance
+		tag  string
+		err  error
+	}{
+		{
+			name: "empty provenance",
+			prov: BYOBProvenance{
+				provenanceV1: &provenanceV1{
+					prov: &intotoAttestation{
+						StatementHeader: intoto.StatementHeader{},
+						Predicate:       slsa1.ProvenancePredicate{},
+					},
+				},
+			},
+			err: serrors.ErrorInvalidDssePayload,
+		},
+		{
+			name: "resolved dependency uri @ refs/heads/main",
+			prov: BYOBProvenance{
+				provenanceV1: &provenanceV1{
+					prov: &intotoAttestation{
+						StatementHeader: intoto.StatementHeader{},
+						Predicate: slsa1.ProvenancePredicate{
+							BuildDefinition: slsa1.ProvenanceBuildDefinition{
+								ResolvedDependencies: []slsa1.ResourceDescriptor{
+									{
+										URI: "git+https://github.com/kubernetes/kubernetes@refs/heads/main",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			tag: "",
+		},
+		{
+			name: "internalParameters uri @ refs/heads/main",
+			prov: BYOBProvenance{
+				provenanceV1: &provenanceV1{
+					prov: &intotoAttestation{
+						StatementHeader: intoto.StatementHeader{},
+						Predicate: slsa1.ProvenancePredicate{
+							BuildDefinition: slsa1.ProvenanceBuildDefinition{
+								InternalParameters: map[string]interface{}{
+									"GITHUB_REF_TYPE": "branch",
+									"GITHUB_REF":      "refs/heads/main",
+								},
+							},
+						},
+					},
+				},
+			},
+			tag: "",
+		},
+		{
+			name: "resolved dependency uri @ refs/tags/v1.0.0",
+			prov: BYOBProvenance{
+				provenanceV1: &provenanceV1{
+					prov: &intotoAttestation{
+						StatementHeader: intoto.StatementHeader{},
+						Predicate: slsa1.ProvenancePredicate{
+							BuildDefinition: slsa1.ProvenanceBuildDefinition{
+								ResolvedDependencies: []slsa1.ResourceDescriptor{
+									{
+										URI: "git+https://github.com/kubernetes/kubernetes@refs/tags/v1.0.0",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			tag: "refs/tags/v1.0.0",
+		},
+		{
+			name: "internalParameters uri @ ref/tags/v1.0.0",
+			prov: BYOBProvenance{
+				provenanceV1: &provenanceV1{
+					prov: &intotoAttestation{
+						StatementHeader: intoto.StatementHeader{},
+						Predicate: slsa1.ProvenancePredicate{
+							BuildDefinition: slsa1.ProvenanceBuildDefinition{
+								InternalParameters: map[string]interface{}{
+									"GITHUB_REF_TYPE": "tag",
+									"GITHUB_REF":      "refs/tags/v1.0.0",
```

Comment:
i dont remember why we need to fall back to `GITHUB_REF`. Why is reliance on resolvedDep only not correct?

Reply:
This was mostly to ensure backwards compatibility with the previous code. I kept it mostly to be safe, but I agree that we shouldn't need to keep it.

Reply:
since we have not released any v1.0 spec support, do we need backward compatibility? Or can we *not* fall back to using the GITHUB_REF`?

Reply:
Well we might still need to do it for v0.2. But you're strictly right that for v1.0 we don't need the backward's compatibility.

Reply:
and for v0.2, where do we need it? npm?

Do our regression tests fail if we don't use `GITHUB_REF`?

Reply:
For npm no, but maybe(?) for others.

Reply:
all our builders generate the resolvedDeps, from the first release, afaik

Reply:
SG, I think we can just remove that code then.

Reply:
SG. Trying to not take on technical debt.

Reply:
done.